flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/rules/physical/batch/BatchExecWindowAggregateRule.scala

```diff
@@ -70,6 +71,11 @@ class BatchExecWindowAggregateRule
   extends RelOptRule(
     operand(classOf[FlinkLogicalWindowAggregate],
       operand(classOf[RelNode], any)),
+    RelBuilder.proto(
+      Contexts.of(
+        RelFactories.DEFAULT_STRUCT,
+        RelBuilder.Config.DEFAULT
+          .withPruneInputOfAggregate(false))),
```

Comment:
Introduce Flink`s own RelBuilderFactory would save such magician codes

Reply:
`RelBuilder.proto` is already a tool method to generate the `RelBuilderFactory`, i don't see much necessity to introduce another interface.